clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -302,27 +302,20 @@ private long sendProducerData(long now) {
     }
 
     private boolean maybeSendTransactionalRequest(long now) {
-        if (transactionManager.isCompletingTransaction() &&
-                !transactionManager.hasPartitionsToAdd() &&
```

Comment:
The check was needed because of the return below when there are still unflushed batches remaining. Now that that check is gone, it seems unnecessary. Maybe @apurvam can confirm if that's correct.